integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/HighlighterFastVectorIT.java

```diff
@@ -0,0 +1,303 @@
+								"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's",
+								" style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void boundaryScannerWordUsingLambda() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.fastVector()
+						.boundaryScanner( bs -> bs.word().locale( Locale.ENGLISH ) )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList(
+								"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's",
+								" style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void boundaryScannerSentence() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.fastVector()
+						.boundaryScanner()
+						.sentence()
+						.locale( Locale.ENGLISH )
+						.end()
+						// limit the fragment size so that the result is sentence-per-highlighted-fragment.
+						// trying to go with a lower value, e.g. with 50 would result in 3 fragments, where second sentence
+						// gets highlighted two times, once per occurrence :shrug:
+						// Result with fragment size  == 50
+						//    "Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. ",
+						//    "Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft rock. ",
+						//    "Since the band's inception, its musical style has ranged from hard rock, heavy metal and glam metal to soft <em>rock</em>."
+						.fragmentSize( "Scorpions are a German rock band formed in Hanover in 1965 by guitarist Rudolf Schenker".length() )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList(
+								"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. ",
+								"Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void styledSchema() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "another" ) )
+				.highlighter( h2 -> h2.fastVector().tagSchema( HighlighterTagSchema.STYLED ) )
```

Comment:
That tag schema is really just a predefined set of tags. Our Lucene implementation works the same as Elasticsearch. If a tag schema is set, then the tags are ignored, and predefined tags from the schema are used. I've added a new test to check for this specifically. There's also a mention of this behaviour in the Javadocs that we have:

> If a tag schema is applied to a highlighter, it will take precedence over {@link #tag(String, String) manually defined tags}.